### PR TITLE
partial front end functionality

### PR DESCRIPTION
**Pull request by cmokbel1**
Created ToDoLists component to render each list that is found in the db for the user in an unordered list. Created multiple components/functions for user to login. Created application functionality to acquire user information, list information for ToDoColumn and singular list information to pass to app.js' listState. refactored certain components to reflect changes in states.  created logout functionality that is reflected in the userState. also added a dependency for the getLists useEffect to fire on userState change. Still need to reflect logouts in the state change which will be committed at a later date.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (1gm) [inline .vscode/settings.json:1]: This file/folder shouldn't be committed. Add it to .gitignore

File: backend/http/server.go [hunk 2]
```diff
@@ -206,7 +206,7 @@ func (s *Server) cors(next http.Handler) http.Handler {
 		headers := w.Header()
 		if s.Domain == "localhost" {
 			// TODO(1gm): Let's make this a config setting for dev environment.
-			headers.Set("Access-Control-Allow-Origin", "http://localhost:5000")
+			headers.Set("Access-Control-Allow-Origin", "http://localhost:3000")
```
Comment (1gm) [inline backend/http/server.go:209]: Undo this change, merge main back into your branch.

File: frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (1gm) [inline frontend/README.md:1]: Add the relevant parts from this to the root README.md - e.g. how to run the app (see what was done for the backend section)

File: frontend/public/index.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
```
Comment (1gm) [inline frontend/public/index.html:8]: This doesn't seem to be formatted consistently, the 'content' value should also be updated.

File: frontend/src/App.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+import './App.css';
+
+import { default as Login } from './components/login';
```
Comment (1gm) [inline frontend/src/App.js:4]: Does `import Login from './components/login'` not work? I haven't seen someone do aliasing for the default export of module before.

File: frontend/src/App.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+import './App.css';
+
```
Comment (1gm) [inline frontend/src/App.js:3]: Maybe group the imports together? Remove the extra white spaces between imports.

File: frontend/src/components/ToDoColumn.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { ToDoLists } from './ToDoLists'
+import { getLists } from '../http/lists'
+const ToDoColumn = ({userState, listState, setListState}) => {
```
Comment (1gm) [inline frontend/src/components/ToDoColumn.js:4]: Should be consistent with your formatting across files. There is no space between the component and the imports, it makes it hard to read and separate the type of code I'm looking at.

File: frontend/src/components/ToDoColumn.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    const [lists, setLists] = useState([])
+
+    useEffect(() => {
+        getLists().then(res => {
+            if (res.length >= 1) {
```
Comment (1gm) [inline frontend/src/components/ToDoColumn.js:9]: Does it matter if the res.length is 0?

File: frontend/src/components/ToDoLists.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+export const ToDoLists = ({lists, listState, setListState}) => {
+// when the list button is clicked the API will return a list item and
+// we want to set that list item to a state which will then be passed up
+// this will allow us to render the current list item onto the main page
+    const handleListClick = async(id) => {
+        const list = await getList(id);
+        if (list.ok) {
+            setListState(list)
+        } else {
+            return list.error
```
Comment (1gm) [inline frontend/src/components/ToDoLists.js:13]: Why are we returning list.error? The result of this function shouldn't be a return should it? If we fail to get a list we should set some sort of state, no?

Also, reading that we are passing the state back up makes me think maybe we should have a component that wraps both the todo lists and the todo detail component. Just food for though, no need to do it in this PR.

File: frontend/src/components/ToDoLists.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        }
+    }
+
+    return (
+        <div>
+            <ul className="list-group">
+                {lists.map((list,index) => <li className="list-group-item" key={index}><button className="btn" onClick={() => handleListClick(list.id)}>{list.name}</button></li>)}
```
Comment (1gm) [inline frontend/src/components/ToDoLists.js:20]: Personally not a fan of cramming everything on a single line.

File: frontend/src/components/login.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { loginUser } from '../http/user'
+
+const Login = ({ userState, setUserState }) => {
+
+    const [userInput, setUserInput] = useState('');
+    const [passwordInput, setPasswordInput] = useState('');
+
```
Comment (1gm) [inline frontend/src/components/login.js:8]: Remove this whitespace

File: frontend/src/components/login.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    const [errorState, setErrorState] = useState('');
+
+
+    const handleLogin = event => {
+        event.preventDefault();
+        loginUser(userInput, passwordInput).then(res => {
+            if (res.name) {
+                 setUserState(res.name);
```
Comment (1gm) [inline frontend/src/components/login.js:16]: This code is not formatted

File: frontend/src/components/login.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+                    <label className="input-text" htmlFor="username">username</label>
+                    <input type="text" id="username" name="username" className="form-control" aria-label="username-input" aria-describedby="username-input" onChange={(e) => setUserInput(e.target.value)} />
+                </div>
+                <div className="input col-sm-8 mb-3">
+                    <label className="input-text" htmlFor="password">password</label>
+                    <input type="password" id="password" name="password" className="form-control" aria-label="password-input" aria-describedby="password-input" onChange={(e) => setPasswordInput(e.target.value)} />
+                </div>
+                <p style={{color: 'red'}}>{errorState}</p>
+                <button type="submit" className="btn btn-primary" onClick={handleLogin} disabled={userInput === "" || passwordInput === ""}>Login</button>
+
```
Comment (1gm) [inline frontend/src/components/login.js:39]: Excess white space can be removed

**Review by 1gm — COMMENTED** [review]
Please address the actionable feedback - otherwise looks good for an initial setup

**Review by cmokbel1 — COMMENTED** [review]
I believe I've addressed all the recommended changes

**Review by 1gm — APPROVED** [review]
